### PR TITLE
add getter for Circuit map

### PR DESCRIPTION
**Pull request by kayla90**
getter to return CircuitSettingsMap info

### CODE REVIEW

File: hystrix/settings.go [hunk 1]
```diff
@@ -101,3 +101,11 @@ func getSettings(name string) *settings {
 
 	return s
 }
+
+func GetCircuitSettingsMap() map[string]*settings{
```
Comment (afex) [inline hystrix/settings.go]: exported functions should only return exported types.  in this case, `Settings` should also be exported.

File: hystrix/settings.go [hunk 2]
```diff
@@ -101,3 +101,11 @@ func getSettings(name string) *settings {
...
+	copy := make(map[string]*settings)
+	for key, val := range circuitSettings {
```
Comment (afex) [inline hystrix/settings.go]: you should grab a read lock around the map before iterating.

`settingsMutex.RLock()`

**Comment by kayla90** [comment]
@afex Make fixes according to comments. (1) export Settings structure and (2) grant lock for map reading

File: hystrix/settings.go [hunk 3]
```diff
@@ -101,3 +101,15 @@ func getSettings(name string) *settings {
 
 	return s
 }
+
+func GetCircuitSettingsMap() map[string]*Settings {
```
Comment (afex) [inline hystrix/settings.go]: the word `map` in the function name is redundant, since that is apparent from the return type.

**Comment by afex** [comment]
this looks fine. once you add a test it should be ready to merge

**Comment by kayla90** [comment]
@afex Update the unit test, please help review and check in. Thanks!

**Pull request merged**